src/repo-query/pages/indexing.tsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from "react";
+import { RepoQueryPages } from "../components/Dialog";
+import { REPO_QUERY_COLLECTION_ENDPOINT, REPO_QUERY_EMBED_ENDPOINT } from "../../constants";
+
+export const IndexingPage = ({ ownerName, repoName, setCurrentPage }: { ownerName: string, repoName: string, setCurrentPage: (page: RepoQueryPages) => void }) => {
+    const [statusMessage, setStatusMessage] = useState("We are checking if this repository is indexed. This may take a while.");
+
+    useEffect(() => {
+        async function checkIndexingStatus () {
+            const status = await fetch(`${REPO_QUERY_COLLECTION_ENDPOINT}?owner=${ownerName}&name=${repoName}&branch=beta`);
+            const statusText = await status.text();
+
+            if (statusText.includes("not indexed")) {
```

Comment:
```suggestion
            const response = await fetch(`${REPO_QUERY_COLLECTION_ENDPOINT}?owner=${ownerName}&name=${repoName}&branch=beta`);
            
            if (response.status !== 200) {
```
Checking for the status code will be more idiomatic and reliable as codes as more consistent.